File: exercises/php/cards/cards.php
<?php

class Cards
{
    public function getCards(): array
    {
        $result = [];
        $deck = [];

        for ($suit = 0; $suit < 4; $suit++) {
            for ($faceValue = 0; $faceValue < 13; $faceValue++) {
                $deck[] = [$suit, $faceValue];
            }
        }

        $cardNumber = 0;
        foreach ($deck as $card) {
            $faceValueName = "";
            switch ($card[1]) {
                case 0:
                    $faceValueName = 'ace';
                    break;
                case 1:
                case 2:
                case 3:
                case 4:
                case 5:
                case 6:
                case 7:
                case 8:
                case 9:
                    $faceValueName = strval($card[1] + 1);
                    break;
                case 10:
                    $faceValueName = "jack";
                    break;
                case 11:
                    $faceValueName = "queen";
                    break;
                case 12:
                    $faceValueName = "king";
                    break;
                default:
                    throw new Exception("Something went wrong " . $card[1] . " is not a valid faceValue!");
            }

            $suitName = "";
            switch ($card[0]) {
                case 0:
                    $suitName = "clubs";
                    break;
                case 1:
                    $suitName = "diamonds";
                    break;
                case 2:
                    $suitName = "hearts";
                    break;
                case 3:
                    $suitName = "spades";
                    break;
                default:
                    throw new Exception("Something went wrong " . $card[0] . " is not a valid suitName!");
            }

            $result[$cardNumber] = $faceValueName . " of " . $suitName;
            $cardNumber++;
        }

        return $result;
    }

}

$cards = new Cards();
$deckInOrder = $cards->getCards();

foreach ($deckInOrder as $card) {
    echo $card . PHP_EOL;
}

File: exercises/kotlin/src/main/kotlin/cards/Cards.kt
package cards

class Cards {
    val cards: MutableList<String>
        get() {
            val result = MutableList(52) {""}
            val deck = Array(52) { IntArray(2) }
            for (suit in 0..3) {
                for (faceValue in 0..12) {
                    deck[suit * 13 + faceValue] = intArrayOf(suit, faceValue)
                }
            }
            for ((cardNumber, card) in deck.withIndex()) {
                val faceValueName: String = when (card[1]) {
                    0 -> "ace"
                    1, 2, 3, 4, 5, 6, 7, 8, 9 -> (card[1] + 1).toString()
                    10 -> "jack"
                    11 -> "queen"
                    12 -> "king"
                    else -> throw IllegalArgumentException("Something went wrong " + card[1] + "is not a valid faceValue!")
                }
                val suitName: String = when (card[0]) {
                    0 -> "clubs"
                    1 -> "diamonds"
                    2 -> "hearts"
                    3 -> "spades"
                    else -> throw IllegalArgumentException("Something went wrong " + card[0] + "is not a valid suitName!")
                }
                result[cardNumber] = "$faceValueName of $suitName"
            }
            return result
        }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val cards = Cards()
            val deckInOrder = cards.cards
            for (card in deckInOrder) {
                println(card)
            }
        }
    }
}